react/features/base/lastn/middleware.ts

```diff
@@ -88,7 +88,11 @@ const _updateLastN = debounce(({ dispatch, getState }: IStore) => {
         lastNSelected = 1;
     }
 
-    dispatch(setLastN(lastNSelected));
+    const { lastN } = state['features/base/lastn'];
+
+    if (lastN !== lastNSelected) {
```

Comment:
Will dispatch the action only if there is a change. I'm trying to reduce the number of actions that are dispatched around PARTICIPANT_UPDATE. This way the long chain of middlewares won't be called, there won't be any state updates and the subscriber won't be called. Otherwise I agree that no re-renders will happen but still many many functions will be called. I hope this shaves off some ms from the participant join time.

Reply:
Ack